22_pipeslab/pipeslab.txt

```diff
@@ -1,21 +1,23 @@
 For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
-  Answer: 
+  Answer: ls -l | tail -n +2 | cut -c4 | grep x | wc | cut -c6-7
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
-  Answer: 
+  Answer: ls -l | tail -n +2 | cut -c4,7,10 | grep x | wc | cut -c6-7
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
-  Answer: 
+  Answer: ls -l |tail -n +2 | grep $(date) | wc | cut -c6-7
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
+1: ls -l | tail -n +2 | cut -c1 | grep - | cut -c4 | grep x | wc | cut -c6-7
+2: ls -l | tail -n +2 | cut -c1 | grep - | cut -c4,7,10 | grep x | wc | cut -c6-7
 
-  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
-
+5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
+/sbin/ifconfig | grep 'inet ' | head -1
```

Comment:
I suggest adding `grep -v 127.0.0.1` to the pipeline to remove 127.0.0.1 from the result. Also a `grep -v inet6` is sometimes required to remove IPv6 addressed. In sum you'd want:

```
/sbin/ifconfig | grep inet | grep -v inet6 | grep -v 127.0.0.1 | head -1
```